copy/copy.go

```diff
@@ -1087,6 +1087,18 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 			return types.BlobInfo{}, "", errors.Wrapf(err, "Error trying to reuse blob %s at destination", srcInfo.Digest)
 		}
 		if reused {
+			// Update the BlobInfo with information about how the reused blob is compressed.
+			compressorName := ic.c.blobInfoCache.DigestCompressorName(blobInfo.Digest)
+			if compressorName == internalblobinfocache.UnknownCompression {
+				return types.BlobInfo{}, "", errors.Errorf("Tried to reuse blob %s with unknown compression", blobInfo.Digest)
```

Comment:
… Oh, right, `copy` is using the default cache, which is never `NoCache`; still, it can be in-memory or just empty (`TryReusingBlob` is the first call we make about a blob).

Reply:
Yeah, I was really unsure about treating it as an error.